mesonbuild/msubprojects.py

```diff
@@ -435,6 +439,41 @@ def update(self) -> bool:
             self.wrap.update_hash_cache(self.repo_dir)
         return success
 
+    def syncwrap(self) -> bool:
+        options = T.cast('SyncWrapArguments', self.options)
+        if self.wrap.type != 'git' and not self.wrap.redirected and not os.path.isdir(self.repo_dir):
+            return True
+
+        self.log(f'Sync {self.wrap.name}...')
+        if self.wrap.redirected:
+            self.wrap.filename = str(Path(self.wrap.original_filename).resolve())
+            self.wrap.parse_wrap()
+
+        if self.wrap.type != 'git':
+            return True
+
+        revision = self.wrap.get('revision').strip()
+        if revision.lower() == 'head':
+            return True
+
+        if options.tags:
+            found, new_revision = quiet_git(['describe', '--exact-match', '--tags'], self.repo_dir)
+            if not found:
+                return True
```

Comment:
I would fallback to commit sha1 if there is no tag, and do that by default. Not sure that needs to be an option?

Reply:
Sounds good. Done